pyhms/utils/print_tree.py

```diff
@@ -0,0 +1,34 @@
+import numpy as np
+
+from ..demes.abstract_deme import AbstractDeme
+
+LEVEL_PREFIX = "-- "
+LAST_LEVEL_SYMBOL = "\u2514"
+MIDDLE_LEVEL_SYMBOL = "\u251C"
+
+
+def format_array(solution: np.ndarray, float_format="{:#.2f}") -> str:
+    solution_values = [float_format.format(sol) for sol in solution]
+    return "(" + ", ".join(solution_values) + ")"
+
+
+def format_deme(deme: AbstractDeme, best_fitness: float | None = None) -> str:
+    deme_distinguisher = "***" if best_fitness and deme.best_individual.fitness == best_fitness else ""
+    is_root = deme._sprout_seed is None
+    new_deme_distinguisher = "(new_deme)" if deme.metaepoch_count <= 1 and not is_root else ""
+    sprout = f"spr: {format_array(deme._sprout_seed.genome)};" if not is_root else ""
+    fitness_value = f"f{format_array(deme.best_individual.genome)} = {deme.best_individual.fitness:.2e}"
+    evaluations = f"evals: {deme.n_evaluations}"
+    return f"{deme_distinguisher}{fitness_value} {sprout} {evaluations} {new_deme_distinguisher}"
```

Comment:
This formatting isn't clear for people who don't know the coded insides well. Instead I would expect full words instead of abbreviations. Something like deme info in summary was very good in my opinion. You can consider taking the best out of those two formats. I would suggest something like that:
```
Deme root [0-10] - best fitness: 4.7168e-02
├-- Deme 0 [1-10] - best solution: f(genome) ~= 9.9293e-07, sprouted at genome, evals xxx
├-- Deme 1 [2-10] - best solution: f(genome) ~= 2.0600e-06, sprouted at genome, evals xxx
├-- Deme 2 [3-10] - best solution: f(genome) ~= 2.5203e-06, sprouted at genome, evals xxx
├-- Deme 3 [4-10] - best solution: f(genome) ~= 1.7688e-07, sprouted at genome, evals xxx
```